synapse/cores/common.py

```diff
@@ -1938,3 +1945,70 @@ def _tufosByInetCidr(self, prop, valu, limit=None):
 
         return self.getTufosBy('range', prop, (ipv4addr, ipv4addr+mask), limit=limit)
 
+    #############################################################
+    # support model persistence
+    #############################################################
+
+    def registerModel(self, module):
+        '''
+        Register the given Python module as a Cortex model provider.
+        Also, load the provided types and forms.
+
+        A model provider is an importable Python module with some well-known top-level functions:
+
+          - getInfo(): returns a dict with the required keys:
+            - namespace (str): unique prefix for types and forms added by this module
+            - version (int): the revision number of this model. this *must* increment with every change.
+          - addTypes(synapse.cores.common.Cortex): add any provided types to the given cortex.
+          - addForms(synapse.cores.common.Cortex): add any provided forms to the given cortex.
+          - upgradeModel(synapse.cores.common.Cortex): when invoked, the cortex is using an out-of-date model
+             version, and requests to be updated to the latest version. implementor must manually fix up the types.
+        '''
+        info = module.getInfo()
+        self.formTufoByProp('syn:model',
+                            info.get('namespace'),
+                            module=module.__name__,
+                            **info)
+
+        module.addTypes(self)
```

Comment:
Use 
addTypes = getattr(module,"addTypes",None)
if addTypes != None:
  addTypes(self)

...paradigm for these so modules don't have to implement NOP functions is they don't have things to add.